pkg/statuscodes/statuscodes.go

```diff
@@ -1,16 +1,29 @@
 // Copyright 2022 Outreach Corporation. All Rights Reserved.
 
-// Description: Provides custom status types and helpers
-
-// Package statuscodes provides custom status types and helpers
+// Description: See below
+
+// Package statuscodes is an attempt to create very-high-level buckets/classifications of errors, for two and ONLY two purposes:
+//  1. Categories are intended for super-high-level bucketing of the responsibility for errors, ideally to be used for SLOs/success rate
+//     metrics on dashboards/reporting (availability).
+//  2. Codes are intended for high-level bucketing of categories of errors, so that generic framework-level http/grpc clients can identify
+//     basic things like retriability, without understanding a ton of nuanced error codes.  For example, I was torn between even including
+//     Conflict as a type separate from BadRequest, but decided to because it's still a different level of responsibility over behavior.
+//     We should ideally never need to extend this list, as extensions should be done via wrapped errors specific to the service.
+//
+// For sending service-specific errors, please wrap one of these errors into a more specific error type with your service-specific errors.
+// For example, bad_request.go has been added to this package, which wraps the basic BadRequest status code with more detailed information
```

Comment:
not sure if the linter will actually complain since it's a comment, but can you format to below 120 (I think that's our lll?)

Reply:
done